fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/ViewHelpers.java

```diff
@@ -182,48 +159,32 @@ public String getVersionLabel(final Graph graph,
      *
      * @param graph the graph
      * @param subject the subject
-     * @return the modification date or null if none exists
+     * @return the modification date if it exists
      */
-    public String getVersionDate(final Graph graph,
-                                 final Node subject) {
-        final Iterator<Triple> objects = getObjects(graph, subject, CREATED_DATE);
-        if (objects.hasNext()) {
-            return objects.next().getObject().getLiteralValue().toString();
-        }
-        return "";
+    public Optional<String> getVersionDate(final Graph graph, final Node subject) {
+        return getValue(graph, subject, CREATED_DATE.asNode());
+    }
+
+    private static Optional<String> getValue(final Graph graph, final Node subject, final Node predicate) {
+        final Iterator<Node> objects = listObjects(graph, subject, predicate);
+        return Optional.ofNullable(objects.hasNext() ? objects.next().getLiteralValue().toString() : null);
     }
 
     /**
      * Get the canonical title of a subject from the graph
      *
      * @param graph the graph
-     * @param subject the subject
+     * @param sub the subject
      * @return canonical title of the subject in the graph
      */
-    public String getObjectTitle(final Graph graph, final Node subject) {
-
-        if (subject == null) {
+    public String getObjectTitle(final Graph graph, final Node sub) {
+        if (sub == null) {
             return "";
         }
-
-        final Property[] properties = new Property[] {RDFS_LABEL, DC_TITLE, DCTERMS_TITLE, SKOS_PREFLABEL};
-
-        for (final Property p : properties) {
-            final Iterator<Triple> objects = getObjects(graph, subject, p);
-
-            if (objects.hasNext()) {
-                return objects.next().getObject().getLiteral().getLexicalForm();
-            }
-        }
-
-        if (subject.isURI()) {
-            return subject.getURI();
-        } else if (subject.isBlank()) {
-            return subject.getBlankNodeLabel();
-        } else {
-            return subject.toString();
-        }
-
+        final Optional<String> title = TITLE_PROPERTIES.stream().map(Property::asNode).flatMap(p -> listObjects(
+                graph, sub, p).toList().stream()).filter(Node::isLiteral).map(Node::getLiteral).map(
+                        LiteralLabel::toString).findFirst();
+        return title.orElse(sub.isURI() ? sub.getURI() : sub.isBlank() ? sub.getBlankNodeLabel() : sub.toString());
```

Comment:
personally, I find it easier to read if the expressions are on separate lines:

    list.stream()
           .map(func1)
           .flatMap(func2)
           .collect(func3);